### DISCUSSION TITLE
current requirements for sinking H0,H1,H2, etc.

### DISCUSSION THREAD

**Discussion by willsalz** [phord/Jarvis#7] [Q&A] [answered]
I'm about to wire up a adafruit HUZZAH to the controller lines, and I was wondering what kind of current to expect to sink. If it's too much for the ESP to handle, I might stick some transistors in the circuit to sink instead.

**Comment by willsalz** [comment, marked as answer]
I just saw #1; I can check out the max current ratings for that Oak board and compare against my HUZZAH.

**Reply by phord** [reply]
I think the Jarvis Desk controller is very sensitive to any drain on these lines, so you probably need to sink very little current here. 

You'll notice I'm also sinking 5v on these 3.3v lines. But that's because the device reportedly is designed to handle 5v inputs, even though the device itself is 3.3v powered.

**Reply by phord** [reply]
I just found in my notes the measured current on each line.  I think I measured this with zero-ohm load between each pin and ground.

        Pin 1 -   0.5ma
        Pin 2 -   4.6ma
        Pin 3 - GND
        Pin 4 -   0.5ma
        Pin 5 - 107  ma
        Pin 6 -   0.5ma
        Pin 7 -   0.5ma
        Pin 8 -   0.5ma

**Reply by willsalz** [reply]
awesome thanks for the details! For some reason I thought the HUZZAH could tolerate 5V input, but it most certainly cannot. Time to go dig up some resistors for make shift level shifter…

**Reply by phord** [reply]
The ESP8266 GPIOs are 5v tolerant, [according to the CEO of Espressif](https://www.facebook.com/groups/1499045113679103/permalink/1731855033731442/).  But the supply voltage is not.  The Digispark module I'm using here has a 5v level converter so it can be USB powered. 

The Huzzah does, too.  So, I think you could wire 5v directly to the pin labeled USB for power; but then you need to be careful to avoid connecting to the USB port when it's already powered.  [Lady Ada does explicitly say that the GPIOs are 3.3v-only](https://learn.adafruit.com/adafruit-feather-huzzah-esp8266/pinouts#gpio-pins-2824936-15) and they are not 5v-tolerant.  So, probably you should heed her advice and use a level shifter.  But then she also links to the [spec sheet](http://www.adafruit.com/datasheets/ESP8266_Specifications_English.pdf) as proof.  And therein, it says:

> All digital IO pins are protected from over-voltage with a snap-back circuit connected between the pad and ground. The snap back voltage is typically about 6V, and the holding voltage is 5.8V. This provides protection from over-voltages and ESD. The output devices are also protected from reversed voltages with diodes.

Since I only use the GPIOs to sink current, I don't care about being able to source 5v or not. So I just wired my ESP8266 GPIOs directly to the desk. But keep in mind that I'm an idiot and this could be a bad idea.

I'm not skilled enough with hardware to keep level-shifters around, so I try to use cheap boards instead.  Huzzah is cheap enough, but I agree it would be annoying if it kept dying on you.
